pkg/infra/state_reader/state_reader.go

```diff
@@ -27,7 +29,8 @@ func NewPulumiReader(templates map[string]statetemplate.StateTemplate, kb knowle
 
 func (p stateReader) ReadState(state []byte, graph construct.Graph) (construct.Graph, error) {
 	returnGraph := construct.NewGraph()
-	internalState, err := p.converter.ConvertState(state)
+	stateReader := bytes.NewReader(state)
```

Comment:
like for the converter, the reader could also use `io.Reader` so that the File passes all the way down

Reply:
oh i see, sure let me change that